tests/test_command_generation.py

```diff
@@ -0,0 +1,347 @@
+
+test_path = os.path.dirname(os.path.abspath(__file__))
+test_output_root = os.path.join(test_path, "tests", "test_output")
+
+
+def test_symlink_entity():
+    """
+    Test the execution of the python script created by symlink_op
+    """
+
+    source = pathlib.Path(test_output_root) / "sym_source"
+    os.mkdir(source)
+
+    # entity_path to be the dest dir
+    entity_path = os.path.join(test_output_root, "entity_name")
+    os.mkdir(entity_path)
+
+    target = pathlib.Path(test_output_root) / "entity_name" / "sym_source"
+
+    # # Build the command
+    cmd = commandgenerator.symlink_op(source, entity_path)
+    # # execute the command
+    subprocess.run([sys.executable, "-c", cmd])
+
+    # # Assert the two files are the same file
+    assert target.is_symlink()
+    assert os.readlink(target) == str(source)
+
+    # Clean up the test directory
+    os.unlink(target)
+    os.rmdir(pathlib.Path(test_output_root) / "entity_name")
+    os.rmdir(pathlib.Path(test_output_root) / "sym_source")
+
+
+def test_copy_op_file():
+    """Test the execution of the python script created by copy_op
+    Test the operation to copy the content of the source file to the destination path
+    with an empty file of the same name already in the directory"""
+
+    to_copy = os.path.join(test_output_root, "to_copy")
+    os.mkdir(to_copy)
+
+    source_file = pathlib.Path(to_copy) / "copy_file.txt"
+    with open(source_file, "w+", encoding="utf-8") as dummy_file:
+        dummy_file.write("dummy")
+
+    entity_path = os.path.join(test_output_root, "entity_name")
+    os.mkdir(entity_path)
+
+    dest_file = os.path.join(test_output_root, "entity_name", "copy_file.txt")
+    with open(dest_file, "w+", encoding="utf-8") as dummy_file:
+        dummy_file.write("")
+
+    # Execute copy
+    cmd = commandgenerator.copy_op(source_file, entity_path)
+    subprocess.run([sys.executable, "-c", cmd])
+
+    # clean up
+    os.remove(pathlib.Path(to_copy) / "copy_file.txt")
+    os.rmdir(pathlib.Path(test_output_root) / "to_copy")
+
+    os.remove(pathlib.Path(entity_path) / "copy_file.txt")
+    os.rmdir(pathlib.Path(test_output_root) / "entity_name")
+
+
+def test_copy_op_dirs():
+    """Test the execution of the python script created by copy_op
+    Test the oeprations that copies an entire directory tree source to a new location destination
+    """
+
+    to_copy = os.path.join(test_output_root, "to_copy")
+    os.mkdir(to_copy)
+
+    # write some test files in the dir
+    source_file = pathlib.Path(to_copy) / "copy_file.txt"
+    with open(source_file, "w+", encoding="utf-8") as dummy_file:
+        dummy_file.write("dummy1")
+
+    source_file_2 = pathlib.Path(to_copy) / "copy_file_2.txt"
+    with open(source_file_2, "w+", encoding="utf-8") as dummy_file:
+        dummy_file.write("dummy2")
+
+    # entity_path to be the dest dir
+    entity_path = os.path.join(test_output_root, "entity_name")
+    os.mkdir(entity_path)
+    # copy those files?
```

Comment:
there is a ? mark here!! Need to ask a Q during core?

Reply:
removed